File: apps/java/src/main/java/tdd/fizzbuzz/domain/type/FizzBuzzType01.java
package tdd.fizzbuzz.domain.type;

import tdd.fizzbuzz.domain.model.FizzBuzzValue;

public final class FizzBuzzType01 extends FizzBuzzType {

    @Override
    public FizzBuzzValue generate(int number) {
        if (isFizzBuzz(number)) {
            return new FizzBuzzValue(number, "FizzBuzz");
        }
        if (isFizz(number)) {
            return new FizzBuzzValue(number, "Fizz");
        }
        if (isBuzz(number)) {
            return new FizzBuzzValue(number, "Buzz");
        }
        return new FizzBuzzValue(number, Integer.toString(number));
    }
}

File: apps/ruby/lib/fizz_buzz/domain/type/fizz_buzz_type_01.rb
# frozen_string_literal: true

class FizzBuzzType01 < FizzBuzzType
  def generate(number)
    return FizzBuzzValue.new('FizzBuzz', number) if fizz?(number) && buzz?(number)
    return FizzBuzzValue.new('Fizz', number) if fizz?(number)
    return FizzBuzzValue.new('Buzz', number) if buzz?(number)

    FizzBuzzValue.new(number.to_s, number)
  end
end
